docs/migration_guides/algorithms_migration.rst

```diff
@@ -0,0 +1,790 @@
+        b. **Probability distribution** (from sampling the device) - you will need a ``Sampler``
+
+    2. How do you want to execute your circuits?
+        a. Using **local** statevector simulators for quick prototyping: **Reference Primitives**
+        b. Using **local** noisy simulations for finer algorithm tuning: **Aer Primitives**
+        c. Accessing **runtime-enabled backends** (or cloud simulators): **Runtime Primitives**
+        d. Accessing **non runtime-enabled backends** : **Backend Primitives**
+
+    For more information and examples, see the `Quantum Instance migration guide <https://qisk.it/qi_migration>`_.
+
+Minimum Eigensolvers
+--------------------
+*Back to* `TL;DR`_
+
+Instead of a :class:`~qiskit.utils.QuantumInstance`, :mod:`qiskit.algorithms.minimum_eigensolvers` are now initialized
+using an instance of the :mod:`~qiskit.primitives.Sampler` or :mod:`~qiskit.primitives.Estimator` primitive, depending
+on the algorithm.
+
+.. attention::
+
+    For the :mod:`qiskit.algorithms.minimum_eigensolvers` classes, depending on the import path,
+    you will access either the primitive-based or the quantum-instance-based
+    implementation. You have to be extra-careful, because the class name does not change.
+
+    * Old import path (Quantum Instance): ``from qiskit.algorithms import VQE, QAOA, NumPyMinimumEigensolver``
+    * New import path (Primitives): ``from qiskit.algorithms.minimum_eigensolvers import VQE, SamplingVQE, QAOA, NumPyMinimumEigensolver``
+
+VQE
+~~~
+
+The legacy :class:`qiskit.algorithms.minimum_eigen_solvers.VQE` class has now been split according to the use-case:
+
+- For general-purpose hamiltonians, you can use the Estimator-based :class:`qiskit.algorithms.minimum_eigensolvers.VQE`
+  class.
+- If you have a diagonal hamiltonian, and would like the algorithm to return a sampling of the state, you can use
+  the new Sampler-based :class:`qiskit.algorithms.minimum_eigensolvers.SamplingVQE` algorithm. This could formerly
+  be realized using the legacy :class:`~qiskit.algorithms.minimum_eigen_solvers.VQE` with
+  :class:`~qiskit.opflow.expectations.CVaRExpectation`.
+
+.. note::
+
+    The new :class:`~qiskit.algorithms.minimum_eigensolvers.VQEResult` class does not include the state anymore, as
+    this output was only useful in the case of diagonal operators. However, if it is available as part of the new
+    :class:`~qiskit.algorithms.minimum_eigensolvers.SamplingVQE` :class:`~qiskit.algorithms.minimum_eigensolvers.SamplingVQEResult`.
+
+.. raw:: html
+
+    <details>
+    <summary><a><font size="+1">VQE Example</font></a></summary>
+    <br>
+
+**[Legacy] Using Quantum Instance:**
+
+.. code-block:: python
+
+    from qiskit.algorithms import VQE
+    from qiskit.algorithms.optimizers import SLSQP
+    from qiskit.circuit.library import TwoLocal
+    from qiskit.opflow import PauliSumOp
+    from qiskit_aer import AerSimulator
+
+    ansatz = TwoLocal(2, 'ry', 'cz')
+    opt = SLSQP(maxiter=1000)
```

Comment:
SPSA or something else where there is noise (shot noise in this case). SLSQP, using finite diff gradient usually will not get anywhere.

Reply:
Updated both examples to use SPSA